htdocs/class/xoopsform/formtabtray.php

```
@@ -0,0 +1,342 @@
+
+        return $this->_currentTab;
+    }
+
+    /**
+     * Add an element to the current tab. If no tab has been started yet, an
+     * untitled one is created so the element is never lost.
+     *
+     * @param XoopsFormElement $formElement element to add
+     * @param bool             $required    mark the element required?
+     */
+    public function addElement(XoopsFormElement $formElement, $required = false)
```

PHPMD (2.15.0)

[error] The method addElement has a boolean flag argument $required, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)